### PR TITLE
Update RBAC for leader election of `machine-controller-manager`

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform openstack

**What this PR does / why we need it**:
Update RBAC for leader election of `machine-controller-manager` before vendoring MCM `v0.43.0`.

**Which issue(s) this PR fixes**:
Fixes part of gardener/gardener#4742

**Special notes for your reviewer**:
Related to gardener/machine-controller-manager#662

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by kon-angelo — APPROVED** [review]
/lgtm

File: charts/internal/machine-controller-manager/seed/templates/clusterrole.yaml [hunk 1]
```diff
@@ -26,3 +26,19 @@ rules:
...
+  resources:
+  - leases
+  verbs:
+  - create
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  resourceNames:
+  - machine-controller-manager
```
Comment (shafeeqes) [inline charts/internal/machine-controller-manager/seed/templates/clusterrole.yaml:40]: we probably need `machine-controller` resourceName here too. I missed it.
@kon-angelo

Comment (shafeeqes) [reply charts/internal/machine-controller-manager/seed/templates/clusterrole.yaml:40]: We can add it while we update the image for the machine-controller-manager-provider-openstack too.

Comment (kon-angelo) [reply charts/internal/machine-controller-manager/seed/templates/clusterrole.yaml:40]: You mean to make a new release of mcm-provider-openstack with the v0.43 MCM changes ?

No objection, but if it is something that can be fixed immediately with an extra PR, I am not sure if we should wait for the mcm-provider-openstack release.

Comment (shafeeqes) [reply charts/internal/machine-controller-manager/seed/templates/clusterrole.yaml:40]: But only with the mcm-provider-openstack release with revendored MCM v0.43.0, we need rbac for the `machine-controller` resource.
I can open a PR in gardener-extension-provider-openstack once there is a new release of the same.

**Pull request merged**